File: Syncfusion_Sample/Windows/Calculate.Windows/Samples/2.0/Working With CalcQuick Demo/VB/MoreComplexForm.vb
#Region "Copyright Syncfusion Inc. 2001 - 2006"
'
'  Copyright Syncfusion Inc. 2001 - 2006. All rights reserved.
'
'  Use of this code is subject to the terms of our license.
'  A copy of the current license can be obtained at any time by e-mailing
'  licensing@syncfusion.com. Any infringement will be prosecuted under
'  applicable laws. 
'
#End Region

Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms

Imports Syncfusion.Calculate

Public Class MoreComplexForm
    Inherits System.Windows.Forms.Form
    Private label1 As System.Windows.Forms.Label
    Private label2 As System.Windows.Forms.Label
    Private label3 As System.Windows.Forms.Label
    Private label4 As System.Windows.Forms.Label
    Private label5 As System.Windows.Forms.Label
    Private label6 As System.Windows.Forms.Label
    Private label7 As System.Windows.Forms.Label



    Private Total As System.Windows.Forms.TextBox
    Private Price As System.Windows.Forms.TextBox
    Private itemComboBox As System.Windows.Forms.ComboBox
    Private Quantity As System.Windows.Forms.TextBox
    Private Shipping As System.Windows.Forms.TextBox
    Private Tax As System.Windows.Forms.TextBox
    Private Discount As System.Windows.Forms.ComboBox

    Private components As System.ComponentModel.Container = Nothing


    Public Sub New()
        '
        ' Required for Windows Form Designer support.
        '
        InitializeComponent()
        Try
            Dim ico As Icon = New Icon(SelectForm.FindFile("Icon", "sfgrid.ico"))
            Me.Icon = ico
        Catch
        End Try
    End Sub 'New


    Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
        If disposing Then
            If Not (components Is Nothing) Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(disposing)
    End Sub 'Dispose

#Region "Windows Form Designer generated code"

    '/ <summary>
    '/ Required method for Designer support; do not modify
    '/ the contents of this method with the code editor.
    '/ </summary>
    Private Sub InitializeComponent()
        Me.label1 = New System.Windows.Forms.Label
        Me.label2 = New System.Windows.Forms.Label
        Me.label3 = New System.Windows.Forms.Label
        Me.label4 = New System.Windows.Forms.Label
        Me.label5 = New System.Windows.Forms.Label
        Me.label6 = New System.Windows.Forms.Label
        Me.label7 = New System.Windows.Forms.Label
        Me.Quantity = New System.Windows.Forms.TextBox
        Me.Shipping = New System.Windows.Forms.TextBox
        Me.Tax = New System.Windows.Forms.TextBox
        Me.Total = New System.Windows.Forms.TextBox
        Me.Price = New System.Windows.Forms.TextBox
        Me.itemComboBox = New System.Windows.Forms.ComboBox
        Me.Discount = New System.Windows.Forms.ComboBox
        Me.SuspendLayout()
        '
        'label1
        '
        Me.label1.Font = New System.Drawing.Font("Microsoft Sans Serif", 8.25!, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.label1.Location = New System.Drawing.Point(56, 40)
        Me.label1.Name = "label1"
        Me.label1.Size = New System.Drawing.Size(64, 23)
        Me.label1.TabIndex = 7
        Me.label1.Text = "Item"
        Me.label1.TextAlign = System.Drawing.ContentAlignment.MiddleRight
        '
        'label2
        '
        Me.label2.Font = New System.Drawing.Font("Microsoft Sans Serif", 8.25!, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.label2.Location = New System.Drawing.Point(56, 80)
        Me.label2.Name = "label2"
        Me.label2.Size = New System.Drawing.Size(64, 23)
        Me.label2.TabIndex = 8
        Me.label2.Text = "Quantity"
        Me.label2.TextAlign = System.Drawing.ContentAlignment.MiddleRight
        '
        'label3
        '
        Me.label3.Font = New System.Drawing.Font("Microsoft Sans Serif", 8.25!, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.label3.Location = New System.Drawing.Point(56, 120)
        Me.label3.Name = "label3"
        Me.label3.Size = New System.Drawing.Size(64, 23)
        Me.label3.TabIndex = 9
        Me.label3.Text = "Price"
        Me.label3.TextAlign = System.Drawing.ContentAlignment.MiddleRight
        '
        'label4
        '
        Me.label4.Font = New System.Drawing.Font("Microsoft Sans Serif", 8.25!, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.label4.Location = New System.Drawing.Point(56, 160)
        Me.label4.Name = "label4"
        Me.label4.Size = New System.Drawing.Size(64, 23)
        Me.label4.TabIndex = 10
        Me.label4.Text = " Discount"
        Me.label4.TextAlign = System.Drawing.ContentAlignment.MiddleRight
        '
        'label5
        '
        Me.label5.Font = New System.Drawing.Font("Microsoft Sans Serif", 8.25!, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.label5.Location = New System.Drawing.Point(56, 200)
        Me.label5.Name = "label5"
        Me.label5.Size = New System.Drawing.Size(64, 23)
        Me.label5.TabIndex = 11
        Me.label5.Text = "Shipping"
        Me.label5.TextAlign = System.Drawing.ContentAlignment.MiddleRight
        '
        'label6
        '
        Me.label6.Font = New System.Drawing.Font("Microsoft Sans Serif", 8.25!, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.label6.Location = New System.Drawing.Point(56, 240)
        Me.label6.Name = "label6"
        Me.label6.Size = New System.Drawing.Size(64, 23)
        Me.label6.TabIndex = 12
        Me.label6.Text = "Tax"
        Me.label6.TextAlign = System.Drawing.ContentAlignment.MiddleRight
        '
        'label7
        '
        Me.label7.Font = New System.Drawing.Font("Microsoft Sans Serif", 8.25!, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.label7.Location = New System.Drawing.Point(56, 280)
        Me.label7.Name = "label7"
        Me.label7.Size = New System.Drawing.Size(64, 23)
        Me.label7.TabIndex = 13
        Me.label7.Text = "Total"
        Me.label7.TextAlign = System.Drawing.ContentAlignment.MiddleRight
        '
        'Quantity
        '
        Me.Quantity.Location = New System.Drawing.Point(128, 80)
        Me.Quantity.Name = "Quantity"
        Me.Quantity.Size = New System.Drawing.Size(100, 20)
        Me.Quantity.TabIndex = 1
        '
        'Shipping
        '
        Me.Shipping.Location = New System.Drawing.Point(128, 200)
        Me.Shipping.Name = "Shipping"
        Me.Shipping.Size = New System.Drawing.Size(100, 20)
        Me.Shipping.TabIndex = 4
        '
        'Tax
        '
        Me.Tax.Location = New System.Drawing.Point(128, 240)
        Me.Tax.Name = "Tax"
        Me.Tax.Size = New System.Drawing.Size(100, 20)
        Me.Tax.TabIndex = 5
        '
        'Total
        '
        Me.Total.Location = New System.Drawing.Point(128, 280)
        Me.Total.Name = "Total"
        Me.Total.Size = New System.Drawing.Size(100, 20)
        Me.Total.TabIndex = 6
        Me.Total.Text = "= ([Quantity] * [Price]) * (1-[Discount]) + [Shipping] + [Tax]"
        '
        'Price
        '
        Me.Price.Location = New System.Drawing.Point(128, 120)
        Me.Price.Name = "Price"
        Me.Price.Size = New System.Drawing.Size(100, 20)
        Me.Price.TabIndex = 2
        Me.Price.Text = "0"
        '
        'itemComboBox
        '
        Me.itemComboBox.DropDownStyle = System.Windows.Forms.ComboBoxStyle.DropDownList
        Me.itemComboBox.Items.AddRange(New Object() {"WidgetA", "WidgetB", "WidgetC"})
        Me.itemComboBox.Location = New System.Drawing.Point(128, 40)
        Me.itemComboBox.Name = "itemComboBox"
        Me.itemComboBox.Size = New System.Drawing.Size(104, 21)
        Me.itemComboBox.TabIndex = 0
        '
        'Discount
        '
        Me.Discount.DropDownStyle = System.Windows.Forms.ComboBoxStyle.DropDownList
        Me.Discount.Items.AddRange(New Object() {"1%", "2%", "3%", "5%", "10%", "20%"})
        Me.Discount.Location = New System.Drawing.Point(128, 160)
        Me.Discount.Name = "Discount"
        Me.Discount.Size = New System.Drawing.Size(121, 21)
        Me.Discount.TabIndex = 3
        '
        'MoreComplexForm
        '
        Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
        Me.BackColor = System.Drawing.Color.FromArgb(CType(CType(237, Byte), Integer), CType(CType(240, Byte), Integer), CType(CType(246, Byte), Integer))
        Me.ClientSize = New System.Drawing.Size(344, 350)
        Me.Controls.Add(Me.Discount)
        Me.Controls.Add(Me.itemComboBox)
        Me.Controls.Add(Me.Price)
        Me.Controls.Add(Me.Total)
        Me.Controls.Add(Me.Tax)
        Me.Controls.Add(Me.Shipping)
        Me.Controls.Add(Me.Quantity)
        Me.Controls.Add(Me.label7)
        Me.Controls.Add(Me.label6)
        Me.Controls.Add(Me.label5)
        Me.Controls.Add(Me.label4)
        Me.Controls.Add(Me.label3)
        Me.Controls.Add(Me.label2)
        Me.Controls.Add(Me.label1)
        Me.Name = "MoreComplexForm"
        Me.StartPosition = System.Windows.Forms.FormStartPosition.CenterScreen
        Me.Text = "More Complex Form Demo"
        Me.ResumeLayout(False)
        Me.PerformLayout()

    End Sub 'InitializeComponent 
#End Region


    Private calculator As CalcQuick = Nothing


    Private Sub MoreComplexForm_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Dim s As String = Me.Quantity.Name

        'Initial settings done in the designer.

        'Instantiate a CalcQuick object:
        calculator = New CalcQuick()

        'Register the controls used in calculations.
        'The formula names used are the Control.Name strings.
        Me.calculator.RegisterControlArray(New Control() { _
                     Me.Quantity, _
                     Me.Price, _
                     Me.Discount, _
                     Me.Shipping, _
                     Me.Tax, _
                     Me.Total _
                    })
        'Allow the CalcQuick sheet to create dependency lists among the formula objects
        'necesary for auto-calculations and do the initial computations.
        Me.calculator.RefreshAllCalculations()
    End Sub 'MoreComplexForm_Load
End Class 'MoreComplexForm

File: Syncfusion_Sample/Windows/Calculate.Windows/Samples/2.0/Working With CalcQuick Demo/CS/MoreComplexForm.cs
#region Copyright Syncfusion Inc. 2001 - 2006
//
//  Copyright Syncfusion Inc. 2001 - 2006. All rights reserved.
//
//  Use of this code is subject to the terms of our license.
//  A copy of the current license can be obtained at any time by e-mailing
//  licensing@syncfusion.com. Any infringement will be prosecuted under
//  applicable laws. 
//
#endregion

using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;

using Syncfusion.Calculate;

namespace FirstSample
{
	/// <summary>
	/// Summary description for MoreComplexForm.
	/// </summary>
	public class MoreComplexForm : System.Windows.Forms.Form
	{
		private System.Windows.Forms.Label label1;
		private System.Windows.Forms.Label label2;
		private System.Windows.Forms.Label label3;
		private System.Windows.Forms.Label label4;
		private System.Windows.Forms.Label label5;
		private System.Windows.Forms.Label label6;
		private System.Windows.Forms.Label label7;
		
		
		
		private System.Windows.Forms.TextBox Total;
		private System.Windows.Forms.TextBox Price;
		private System.Windows.Forms.ComboBox itemComboBox;
		private System.Windows.Forms.TextBox Quantity;
		private System.Windows.Forms.TextBox Shipping;
		private System.Windows.Forms.TextBox Tax;
		private System.Windows.Forms.ComboBox Discount;
		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;

		public MoreComplexForm()
		{
			//
			// Required for Windows Form Designer support.
			//
			InitializeComponent();

			try
			{
				System.Drawing.Icon ico = new System.Drawing.Icon(SelectForm.FindFile("Icon", "sfgrid.ico"));
				this.Icon = ico ;
			}
			catch{}
	

			//
			// TODO: Add any constructor code after InitializeComponent call
			//
		}



		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if(components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support; do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
            this.label1 = new System.Windows.Forms.Label();
            this.label2 = new System.Windows.Forms.Label();
            this.label3 = new System.Windows.Forms.Label();
            this.label4 = new System.Windows.Forms.Label();
            this.label5 = new System.Windows.Forms.Label();
            this.label6 = new System.Windows.Forms.Label();
            this.label7 = new System.Windows.Forms.Label();
            this.Quantity = new System.Windows.Forms.TextBox();
            this.Shipping = new System.Windows.Forms.TextBox();
            this.Tax = new System.Windows.Forms.TextBox();
            this.Total = new System.Windows.Forms.TextBox();
            this.Price = new System.Windows.Forms.TextBox();
            this.itemComboBox = new System.Windows.Forms.ComboBox();
            this.Discount = new System.Windows.Forms.ComboBox();
            this.SuspendLayout();
            // 
            // label1
            // 
            this.label1.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.label1.Location = new System.Drawing.Point(56, 40);
            this.label1.Name = "label1";
            this.label1.Size = new System.Drawing.Size(64, 23);
            this.label1.TabIndex = 7;
            this.label1.Text = "Item";
            this.label1.TextAlign = System.Drawing.ContentAlignment.MiddleRight;
            // 
            // label2
            // 
            this.label2.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.label2.Location = new System.Drawing.Point(56, 80);
            this.label2.Name = "label2";
            this.label2.Size = new System.Drawing.Size(64, 23);
            this.label2.TabIndex = 8;
            this.label2.Text = "Quantity";
            this.label2.TextAlign = System.Drawing.ContentAlignment.MiddleRight;
            // 
            // label3
            // 
            this.label3.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.label3.Location = new System.Drawing.Point(56, 120);
            this.label3.Name = "label3";
            this.label3.Size = new System.Drawing.Size(64, 23);
            this.label3.TabIndex = 9;
            this.label3.Text = "Price";
            this.label3.TextAlign = System.Drawing.ContentAlignment.MiddleRight;
            // 
            // label4
            // 
            this.label4.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.label4.Location = new System.Drawing.Point(56, 160);
            this.label4.Name = "label4";
            this.label4.Size = new System.Drawing.Size(64, 23);
            this.label4.TabIndex = 10;
            this.label4.Text = " Discount";
            this.label4.TextAlign = System.Drawing.ContentAlignment.MiddleRight;
            // 
            // label5
            // 
            this.label5.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.label5.Location = new System.Drawing.Point(56, 200);
            this.label5.Name = "label5";
            this.label5.Size = new System.Drawing.Size(64, 23);
            this.label5.TabIndex = 11;
            this.label5.Text = "Shipping";
            this.label5.TextAlign = System.Drawing.ContentAlignment.MiddleRight;
            // 
            // label6
            // 
            this.label6.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.label6.Location = new System.Drawing.Point(56, 240);
            this.label6.Name = "label6";
            this.label6.Size = new System.Drawing.Size(64, 23);
            this.label6.TabIndex = 12;
            this.label6.Text = "Tax";
            this.label6.TextAlign = System.Drawing.ContentAlignment.MiddleRight;
            // 
            // label7
            // 
            this.label7.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.label7.Location = new System.Drawing.Point(56, 280);
            this.label7.Name = "label7";
            this.label7.Size = new System.Drawing.Size(64, 23);
            this.label7.TabIndex = 13;
            this.label7.Text = "Total";
            this.label7.TextAlign = System.Drawing.ContentAlignment.MiddleRight;
            // 
            // Quantity
            // 
            this.Quantity.Location = new System.Drawing.Point(128, 80);
            this.Quantity.Name = "Quantity";
            this.Quantity.Size = new System.Drawing.Size(100, 20);
            this.Quantity.TabIndex = 1;
            // 
            // Shipping
            // 
            this.Shipping.Location = new System.Drawing.Point(128, 200);
            this.Shipping.Name = "Shipping";
            this.Shipping.Size = new System.Drawing.Size(100, 20);
            this.Shipping.TabIndex = 4;
            // 
            // Tax
            // 
            this.Tax.Location = new System.Drawing.Point(128, 240);
            this.Tax.Name = "Tax";
            this.Tax.Size = new System.Drawing.Size(100, 20);
            this.Tax.TabIndex = 5;
            // 
            // Total
            // 
            this.Total.Location = new System.Drawing.Point(128, 280);
            this.Total.Name = "Total";
            this.Total.Size = new System.Drawing.Size(100, 20);
            this.Total.TabIndex = 6;
            this.Total.Text = "= ([Quantity] * [Price]) * (1-[Discount]) + [Shipping] + [Tax]";
            // 
            // Price
            // 
            this.Price.Location = new System.Drawing.Point(128, 120);
            this.Price.Name = "Price";
            this.Price.Size = new System.Drawing.Size(100, 20);
            this.Price.TabIndex = 2;
            this.Price.Text = "0";
            // 
            // itemComboBox
            // 
            this.itemComboBox.DropDownStyle = System.Windows.Forms.ComboBoxStyle.DropDownList;
            this.itemComboBox.Items.AddRange(new object[] {
            "WidgetA",
            "WidgetB",
            "WidgetC"});
            this.itemComboBox.Location = new System.Drawing.Point(128, 40);
            this.itemComboBox.Name = "itemComboBox";
            this.itemComboBox.Size = new System.Drawing.Size(104, 21);
            this.itemComboBox.TabIndex = 0;
            // 
            // Discount
            // 
            this.Discount.DropDownStyle = System.Windows.Forms.ComboBoxStyle.DropDownList;
            this.Discount.Items.AddRange(new object[] {
            "1%",
            "2%",
            "3%",
            "5%",
            "10%",
            "20%"});
            this.Discount.Location = new System.Drawing.Point(128, 160);
            this.Discount.Name = "Discount";
            this.Discount.Size = new System.Drawing.Size(121, 21);
            this.Discount.TabIndex = 3;
            // 
            // MoreComplexForm
            // 
            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.BackColor = System.Drawing.Color.FromArgb(((int)(((byte)(237)))), ((int)(((byte)(240)))), ((int)(((byte)(246)))));
            this.ClientSize = new System.Drawing.Size(344, 350);
            this.Controls.Add(this.Discount);
            this.Controls.Add(this.itemComboBox);
            this.Controls.Add(this.Price);
            this.Controls.Add(this.Total);
            this.Controls.Add(this.Tax);
            this.Controls.Add(this.Shipping);
            this.Controls.Add(this.Quantity);
            this.Controls.Add(this.label7);
            this.Controls.Add(this.label6);
            this.Controls.Add(this.label5);
            this.Controls.Add(this.label4);
            this.Controls.Add(this.label3);
            this.Controls.Add(this.label2);
            this.Controls.Add(this.label1);
            this.Name = "MoreComplexForm";
            this.StartPosition = System.Windows.Forms.FormStartPosition.CenterScreen;
            this.Text = "More Complex Form Demo";
            this.Load += new System.EventHandler(this.MoreComplexForm_Load);
            this.ResumeLayout(false);
            this.PerformLayout();

		}
		#endregion


		CalcQuick calculator = null;

		private void MoreComplexForm_Load(object sender, System.EventArgs e)
		{
			//Initial settings done in the designer.


			//Instantiate a CalcQuick object:
			calculator = new CalcQuick();

			//Register the controls used in calculations.
			//The formula names used are the Control.Name strings.
            this.calculator.RegisterControlArray(new Control[]
                                                {
                                                    this.Quantity,
                                                    this.Price,
                                                    this.Discount,
                                                    this.Shipping,
                                                    this.Tax,
                                                    this.Total
												});

			//Allow the CalcQuick sheet to create dependency lists among the formula objects
			//necesary for auto-calculations and do the initial computations.
			this.calculator.RefreshAllCalculations();
		}
	}
}
